toolkits/nature/packages/nature-facet/js/enhanced-facet.js

```diff
@@ -0,0 +1,114 @@
+import {makeArray} from '@springernature/global-javascript/src/helpers';
+import {Expander} from '@springernature/global-expander/js/expander';
+
+/**
+ * Constants
+ */
+
+const DATA_COMPONENT = 'data-facet-expander';
+
+const selectors = {
+	DATA_COMPONENT: `[${DATA_COMPONENT}]`,
+	FACET: '[data-facet]'
+};
+
+const classNames = {
+	SELECTED: 'c-facet__selected',
+	ELLIPSIS: 'c-facet__ellipsis',
+	CLEAR: 'c-facet__clear-selection',
+	BUTTON_CONTAINER: 'c-facet__button-container',
+	SUBMIT: 'c-facet__submit'
+};
+
+const buttonName = {
+	ALL: 'All',
+	SELECTED: 'selected',
+	CLEAR_SELECTION: 'Clear selection',
+	SUBMIT: 'Apply filters'
+};
+
+const facetTargetAttribute = 'data-facet-target';
+
+const containsCheckableInput = element => {
+	return Boolean(element.querySelector('input[type=checkbox], input[type=radio]'));
+};
+
+const createClearButton = (element, container) => {
+	const button = document.createElement('button');
+	button.classList.add(classNames.CLEAR);
+	button.textContent += buttonName.CLEAR_SELECTION;
+
+	button.addEventListener('click', event => {
+		event.preventDefault();
+		const checkedInputs = element.querySelectorAll('input[type=checkbox]:checked, input[type=radio]:checked');
+
+		checkedInputs.forEach(function (checkedInput) {
+			checkedInput.checked = false;
+		});
+
+		button.closest('form').submit();
```

Comment:
`closest` polyfill?

Reply:
Was trying to find out if the IE11 polyfill bundle contains `closest` but found we might have an issue with the recent polyfill changes. I've reached out to Andy to see if he can help.

Reply:
ive added a note in the readme that this requires polyfill